externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/kyuubi/ui/KyuubiUIUtils.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.kyuubi.ui
+
+import java.net.URL
+import javax.servlet.http.{HttpServlet, HttpServletRequest, HttpServletResponse}
+
+import org.sparkproject.jetty.servlet.{ServletContextHandler, ServletHolder}
+
+// the code forked from `org.apache.spark.ui.JettyUtils`
+object KyuubiUIUtils {
+  def createServletHandler(
+      path: String,
+      servlet: HttpServlet,
+      basePath: String): ServletContextHandler = {
+    val prefixedPath = if (basePath == "" && path == "/") {
+      path
+    } else {
+      (basePath + path).stripSuffix("/")
+    }
+    val contextHandler = new ServletContextHandler
+    val holder = new ServletHolder(servlet)
+    contextHandler.setContextPath(prefixedPath)
+    contextHandler.addServlet(holder, "/")
+    contextHandler
+  }
+
+  def createRedirectHandler(
```

Comment:
Why not just call org.apache.spark.ui.JettyUtils. createRedirectHandler

Reply:
We also need to use reflect if we want to use `org.apache.spark.ui.JettyUtils.createRedirectHandler`, and I'm afriad Spark may change the code in future.

Reply:
this can certainly prevent  the impact of future spark code change？

Reply:
OK removed the forked code, only use reflect